src/transaction/log_replication_atomic.cpp

```diff
@@ -119,6 +119,7 @@ namespace cublog
 		// nested atomic replication
 		assert (false);
 	      }
+	    m_lowest_unapplied_lsa = m_redo_lsa;
```

Comment:
I do not think this is correct because there can be concurrent atomic sequences - from other transactions that the transaction for which the current log record is and whose un-applied log records can have LSA's less that the current log record.
Please correct me if I'm wrong.
I think that the new "lowest unapplied lsa" is the minimum between all currently in progress atomic sequences and `m_redo_lsa`.

Reply:
You are correct, I though this might be the case, I will implement the solution.